File: java/src/test/java/io/binac/leetcode/MaximalRectangleTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class MaximalRectangleTest {
    private static final MaximalRectangle.Solution1 SOLUTION1 = new MaximalRectangle.Solution1();
    private static final MaximalRectangle.Solution2 SOLUTION2 = new MaximalRectangle.Solution2();

    private final char[][] MATRIX1 = {
            {'1', '0', '1', '0', '0'},
            {'1', '0', '1', '1', '1'},
            {'1', '1', '1', '1', '1'},
            {'1', '0', '0', '1', '0'}
    };
    private final int EXPECTED1 = 6;

    private final char[][] MATRIX2 = {{'0'}};
    private final int EXPECTED2 = 0;

    private final char[][] MATRIX3 = {{'1'}};
    private final int EXPECTED3 = 1;

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.maximalRectangle(MATRIX1));
        assertEquals(EXPECTED2, SOLUTION1.maximalRectangle(MATRIX2));
        assertEquals(EXPECTED3, SOLUTION1.maximalRectangle(MATRIX3));
    }


    @Test
    void test2() {
        assertEquals(EXPECTED1, SOLUTION2.maximalRectangle(MATRIX1));
        assertEquals(EXPECTED2, SOLUTION2.maximalRectangle(MATRIX2));
        assertEquals(EXPECTED3, SOLUTION2.maximalRectangle(MATRIX3));
    }
}

File: cpp/tests/MaximalRectangleTest.cpp
#include "MaximalRectangle.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(MaximalRectangle, Solution85_1) {
  auto S85_1 = Solution85_1();

  std::vector<std::vector<char>> Matrix1{{'1', '0', '1', '0', '0'},
                                         {'1', '0', '1', '1', '1'},
                                         {'1', '1', '1', '1', '1'},
                                         {'1', '0', '0', '1', '0'}};
  int const Expected1 = 6;
  EXPECT_EQ(Expected1, S85_1.maximalRectangle(Matrix1));

  std::vector<std::vector<char>> Matrix2{{'0'}};
  int const Expected2 = 0;
  EXPECT_EQ(Expected2, S85_1.maximalRectangle(Matrix2));

  std::vector<std::vector<char>> Matrix3{{'1'}};
  int const Expected3 = 1;
  EXPECT_EQ(Expected3, S85_1.maximalRectangle(Matrix3));
}

TEST(MaximalRectangle, Solution85_2) {
  auto S85_2 = Solution85_2();

  std::vector<std::vector<char>> Matrix1{{'1', '0', '1', '0', '0'},
                                         {'1', '0', '1', '1', '1'},
                                         {'1', '1', '1', '1', '1'},
                                         {'1', '0', '0', '1', '0'}};
  int const Expected1 = 6;
  EXPECT_EQ(Expected1, S85_2.maximalRectangle(Matrix1));

  std::vector<std::vector<char>> Matrix2{{'0'}};
  int const Expected2 = 0;
  EXPECT_EQ(Expected2, S85_2.maximalRectangle(Matrix2));

  std::vector<std::vector<char>> Matrix3{{'1'}};
  int const Expected3 = 1;
  EXPECT_EQ(Expected3, S85_2.maximalRectangle(Matrix3));
}
